src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  const separator = fromFormatCopy.pop();
+  const parts = date.split(separator);
```

Comment:
The logic assumes that the last element of `fromFormat` is a separator, which may not always be the case. This could lead to incorrect splitting of the date string. Consider using a more robust method to determine the separator.